cachito/workers/pkg_manager.py

```diff
@@ -56,7 +59,13 @@ def resolve_gomod_deps(archive_path, request_id):
             'PATH': os.environ.get('PATH', ''),
         }
 
-        run_params = {'env': env, 'cwd': source_dir}
+        run_params = {'env': env, 'cwd': app_source_path}
+
+        for dep_override in dep_overrides:
+            name = dep_override['name']
+            version = dep_override['version']
+            log.info('Applying the gomod override %s@%s', name, version)
+            _run_cmd(('go', 'mod', 'edit', '-replace', f'{name}={name}@{version}'), run_params)
```

Comment:
I think we want to have the ability to override across different names. This would be needed for handling replacing a dependency with a fork done in another host, for example. `github.com/foo/bar` gets replaced with `git.internal.com/foo/bar`.